File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/atcoder_ABC070_B.java
import java.util.Scanner;
import java.util.Arrays;

public class atcoder_ABC070_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] parts = input.split(" ");
        int[] A = new int[parts.length];
        for (int i = 0; i < parts.length; i++) {
            A[i] = Integer.parseInt(parts[i]);
        }
        if (A[1] <= A[2] || A[3] <= A[0]) {
            System.out.println(0);
        } else {
            Arrays.sort(A);
            System.out.println(A[2] - A[1]);
        }
    }
}

File: output/deepseek_r1/avatar/direct_translation/java/python/atcoder_ABC070_B.py
a, b, c, d = map(int, input().split())
if c > b:
    print(0)
elif a > d:
    print(0)
elif a < c:
    print(min(b, d) - c)
else:
    l = sorted([a, b, c, d])
    print(l[2] - l[1])
